source/tokens.rs

```diff
@@ -807,4 +1220,348 @@ mod tests {
+        let output = Some(3);
+
+        assert_eq!(input.slice_index(3), output);
+    }
+
+    #[test]
+    fn case_span_index_does_not_exist() {
+        let input  = Span::new(b"foobar");
+        let output = None;
+
+        assert_eq!(input.slice_index(7), output);
+    }
+
+    #[test]
+    fn case_span_empty_compare() {
+        let input  = Span::new(b"");
+        let output = CompareResult::Incomplete;
+
+        assert_eq!(input.compare(b"foobar"), output);
+    }
+
+    #[test]
+    fn case_span_compare_incomplete() {
+        let input  = Span::new(b"foo");
+        let output = CompareResult::Incomplete;
+
+        assert_eq!(input.compare(b"foobar"), output);
+    }
+
+    #[test]
+    fn case_span_compare_ok() {
+        let input  = Span::new(b"foobar");
+        let output = CompareResult::Ok;
+
+        assert_eq!(input.compare(b"foobar"), output);
+    }
+
+    #[test]
+    fn case_span_empty_compare_no_case() {
+        let input  = Span::new(b"");
+        let output = CompareResult::Incomplete;
+
+        assert_eq!(input.compare(b"foobar"), output);
+    }
+
+    #[test]
+    fn case_span_compare_no_case_incomplete() {
+        let input  = Span::new(b"foo");
+        let output = CompareResult::Incomplete;
+
+        assert_eq!(input.compare(b"foobar"), output);
+    }
+
+    #[test]
+    fn case_span_compare_no_case_ok() {
+        let input  = Span::new(b"foobar");
+        let output = CompareResult::Ok;
+
+        assert_eq!(input.compare(b"foobar"), output);
+    }
+
+    #[test]
+    fn case_span_slice_with_range() {
+        let range  = 2..5;
+        let input  = b"foobar";
+        let output = Span { 
+            offset: 2,
+            line  : 1,
+            column: 3,
+            slice : &input[range.clone()]
+        };
+
+        assert_eq!(Span::new(input).slice(range.clone()), output);
+    }
+
+    #[test]
+    fn case_span_slice_with_range_to() {
+        let range  = 2..;
+        let input  = b"foobar";
+        let output = Span { 
+            offset: 2,
+            line  : 1,
+            column: 3,
+            slice : &input[range.clone()]
+        };
+
+        assert_eq!(Span::new(input).slice(range.clone()), output);
+    }
+
+    #[test]
+    fn case_span_slice_with_range_from() {
+        let range  = ..3;
+        let input  = b"foobar";
+        let output = Span { 
+            offset: 0,
+            line  : 1,
+            column: 1,
+            slice : &input[range.clone()]
+        };
+
+        assert_eq!(Span::new(input).slice(range.clone()), output);
+    }
+
+    #[test]
+    fn case_span_slice_with_range_full() {
+        let range  = ..;
+        let input  = b"foobar";
+        let output = Span { 
+            offset: 0,
+            line  : 1,
+            column: 1,
+            slice : &input[range.clone()]
+        };
+
+        assert_eq!(Span::new(input).slice(range.clone()), output);
+    }
+
+    #[test]
+    fn case_span_in_parser() {
```

Comment:
I wonder if this kind of test doesn't have it's place in an integration test: this doesn't test only the behavior of the library but the interaction between nom and this module. So we take the place of a user wanting to use both of the libs. What do you think?